docs/commands/cp.md

```diff
@@ -0,0 +1,175 @@
+* `--update` - Update cached list of files for the sources (available only with --local)
+* `--no-glob` - Do not expand globs (such as * or ?) (available only with --local)
+* `--force` - Force creating files even if they already exist (available only with --local)
+* `-h`, `--help` - Show the help message and exit
+* `-v`, `--verbose` - Be verbose
+* `-q`, `--quiet` - Be quiet
+
+## Copy Operations
+
+The command supports two main modes of operation:
+
+### Studio Mode (Default)
+When using Studio mode (default), the command copies files and directories through Studio using the configured credentials. This mode automatically determines the operation type based on the source and destination protocols, supporting four different copy scenarios.
+
+### Local Mode
+When using `--local` flag, the command operates directly with local storage access, bypassing Studio. This mode supports additional options like `--anon`, `--update`, `--no-glob`, and `--force`.
+
+## Supported Storage Protocols
+
+The command supports the following storage protocols:
+- **Local file system**: Direct paths (e.g., `/path/to/directory` or `./relative/path`)
+- **AWS S3**: `s3://bucket-name/path`
+- **Google Cloud Storage**: `gs://bucket-name/path`
+- **Azure Blob Storage**: `az://container-name/path`
+
+## Examples
+
+### Studio Mode Examples
+
+The command automatically determines the operation type based on the source and destination protocols:
+
+#### 1. Local to Local (local path → local path)
+**Operation**: Direct local file system copy
+- Uses the local filesystem's native copy operation
+- Fastest operation as no network transfer is involved
+- Supports both files and directories
+
+```bash
+datachain cp /path/to/local/file.txt /path/to/destination/file.txt
+```
+
+#### 2. Local to Remote (local path → `s3://`, `gs://`, `az://`)
+**Operation**: Upload to cloud storage
+- Uploads local files/directories to remote storage
+- Uses presigned URLs for secure uploads
+- Supports S3 multipart form data for large files
+- Requires `--recursive` flag for directories
+
+```bash
+# Upload single file
+datachain cp /path/to/file.txt s3://my-bucket/data/file.txt
+
+# Upload directory recursively
+datachain cp -r /path/to/directory s3://my-bucket/data/
+```
+
+#### 3. Remote to Local (`s3://`, `gs://`, `az://` → local path)
+**Operation**: Download from cloud storage
+- Downloads remote files/directories to local storage
+- Uses presigned download URLs
+- Automatically extracts filename if destination is a directory
+- Creates destination directory if it doesn't exist
+
+```bash
+# Download single file
+datachain cp s3://my-bucket/data/file.txt /path/to/local/file.txt
+
+# Download to directory (filename preserved)
+datachain cp s3://my-bucket/data/file.txt /path/to/directory/
+```
+
+#### 4. Remote to Remote (`s3://` → `s3://`, `gs://` → `gs://`, etc.)
+**Operation**: Copy within cloud storage
+- Copies files between locations in the same bucket
```

Comment:
why this limitation?

Reply:
Because of the limitation in the respective clients. We don't download the files to Studio. We use the s3 endpoints or s3 features to copy file between s3 and so on.